### PR TITLE
Report cfg policy violations to issues

### PR DESCRIPTION
**Pull request by zkdev**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator

```

### CODE REVIEW

**Comment by zkdev** [comment]
/hold
until step adjusted

File: concourse/steps/replicate_secrets.mako [hunk 1]
```diff
@@ -99,11 +103,32 @@ replicate_secrets(
...
+status_reports = cmu.generate_cfg_element_status_reports(
+  cfg_dir='${cfg_repo_relpath}',
+  element_storage='${cfg_repo_url}',
+)
 cmr.create_report(status_reports)
 cfg_report_summary_gen = cmr.cfg_element_statuses_storage_summaries(status_reports)
 cfg_responsible_summary_gen = cmr.cfg_element_statuses_responsible_summaries(status_reports)
 
+% if compliance_reporting_repo_url:
+## use default from `CONCOURSE_CURRENT_CFG` once it refers to proper cfg_set
```
Comment (ccwienk) [inline concourse/steps/replicate_secrets.mako:115]: let us rather keep passing-in repository explicitly at _this_ point. -> no need for TODO/comment.

File: concourse/steps/replicate_secrets.mako [hunk 2]
```diff
@@ -99,11 +103,32 @@ replicate_secrets(
...
+
+report_cfg_policy_status(
+  cfg_factory=cfg_factory,
+  status_reports=status_reports,
+  compliance_reporting_repo_url='${compliance_reporting_repo_url}',
+  delivery_svc_client=delivery_svc_client,
+  github_issue_template_cfgs=${github_issue_template_cfgs},
+)
+% else:
+logger.info('will not generate reporting issues')
```
Comment (ccwienk) [inline concourse/steps/replicate_secrets.mako:129]: ideally also state _why_ (absence of reporting-repo-url) to make the info-message more helpful (please, in general, consider this for logging output; similar to comments in source-code, please always consider expressing WHY over WHAT + non-obvious context-information (think of log-readers that do not have a mapping to the logging source code))

File: cfg_mgmt/util.py [hunk 3]
```diff
@@ -490,3 +492,27 @@ def process_cfg_queue_and_persist_in_repo(
         git_helper.repo.git.reset('--hard', '@~')
 
     return True
+
+
+def scan_result_group_collection(
+    results: tuple[gcm.CfgScanResult],
+):
```
Comment (ccwienk) [inline cfg_mgmt/util.py:499]: nit: might want to add returntypehint (for IDE..) - not super-important, though

**Review by ccwienk — CHANGES_REQUESTED** [review]

File: concourse/steps/replicate_secrets.mako (hunk 1 above)
Comment (zkdev) [reply concourse/steps/replicate_secrets.mako:115]: comment refers to delivery svc

Comment (zkdev) [reply concourse/steps/replicate_secrets.mako]: mved it some lines

File: concourse/steps/replicate_secrets.mako [hunk 4]
```diff
@@ -99,11 +103,32 @@ replicate_secrets(
...
+  element_storage='${cfg_repo_url}',
+)
 cmr.create_report(status_reports)
 cfg_report_summary_gen = cmr.cfg_element_statuses_storage_summaries(status_reports)
 cfg_responsible_summary_gen = cmr.cfg_element_statuses_responsible_summaries(status_reports)
 
+% if compliance_reporting_repo_url:
+delivery_svc_client = ccc.delivery.client(
+  cfg_name='${delivery_endpoints_cfg_name}',
+  ## use default from `CONCOURSE_CURRENT_CFG` once it refers to proper cfg_set
```
Comment (ccwienk) [inline concourse/steps/replicate_secrets.mako:117]: rm comment

**Review by ccwienk — CHANGES_REQUESTED** [review]

**Review by ccwienk — APPROVED** [review]

**Pull request merged**